WebAPI_Unit_Testing/UnitTest1.cs

```diff
@@ -0,0 +1,229 @@
+        }
+
+
+
+        public void LoadJson()
+        {
+            using (StreamReader r = new StreamReader(@"C:\Users\cpadole\source\repos\WebApiDemo_5Sept19\WebApiDemo_5Sept19\Data\BookData.json"))
+            {
+
+                string json = r.ReadToEnd();
+                bookList = JsonConvert.DeserializeObject<List<Book>>(json);
+            }
+        }
+    }
+
+    public class PutMethodTest
+    {
+        List<Book> bookList = new List<Book>();
+
+        [Fact]
+        public void Check_If_Invalid_Object_Is_Passed()
+        {
+
+
+            List<string> serverMessageList = new List<string>() { "Invalid Data is passed" };
+            LoadJson();
+            var expectedBook = new Response(null, serverMessageList);
+            var actualBook = new BookService().put(new Book("Book #1", "Neelesh", -121, 40, "Fiction"));
+            expectedBook.Should().BeEquivalentTo(actualBook);
+        }
+
+        [Fact]
+        public void Check_If_Book_Data_Updated_For_Correct_Id()
+        {
+            List<string> serverMessageList = new List<string>() { "Updated" };
+
+            var expectedBook = new Response(new List<Book>() { new Book("Book #1", "Neelesh", 1, 40, "Non-Fiction") }, serverMessageList);
+            var actualBook = new BookService().put(new Book("Book #1", "Neelesh", 1, 40, "Non-Fiction"));
+            expectedBook.Should().BeEquivalentTo(actualBook);
+        }
+
+        public void LoadJson()
+        {
+            using (StreamReader r = new StreamReader(@"C:\Users\cpadole\source\repos\WebApiDemo_5Sept19\WebApiDemo_5Sept19\Data\BookData.json"))
+            {
+
+                string json = r.ReadToEnd();
+                bookList = JsonConvert.DeserializeObject<List<Book>>(json);
+            }
+        }
+    }
+
+    public class DeleteMethodTest
+    {
+        List<Book> bookList = new List<Book>();
+
+        [Fact]
+        public void Check_For_Id_Entered_Is_Not_Present()
+        {
+            List<string> serverMessageList = new List<string>() { "Item Not Found" };
+            LoadJson();
+            var expectedBook = new Response(bookList, serverMessageList);
+            var actualBook = new BookService().delete(121);
+            expectedBook.Should().BeEquivalentTo(actualBook);
+        }
+
+        [Fact]
+        public void Check_For_Id_Invalid_Id_Entered()
+        {
+            List<string> serverMessageList = new List<string>() { "Invalid Id, Id should be a positive number" };
+            LoadJson();
+            var expectedBook = new Response(null, serverMessageList);
+            var actualBook = new BookService().delete(-124);
+            expectedBook.Should().BeEquivalentTo(actualBook);
+        }
+
+
+
+        [Fact]
+        public void Check_If_Book_Is_Deleted()
+        {
+
+
+            List<string> serverMessageList = new List<string>() { "Item Deleted" };
+
+            //LoadJson();
+
+
+            var actualBook = new BookService().delete(5);
```

Comment:
When u are returning some value then 'var' should be replaced by type .... when initializing with particualar class or list as above 'List' u can use var